File: 프로그래머스/2/42587. 프로세스/프로세스.js
function isHigherPriority(queue, currentPriority) {
    return queue.some(process => process.priority > currentPriority);
}

function solution(priorities, location) {
    let answer = 0;
    const queue = priorities.map((priority, index) => ({ priority, index }));
    
    while (queue.length > 0) {
        const currentProcess = queue.shift();
        if (isHigherPriority(queue, currentProcess.priority)) {
            queue.push(currentProcess);
        } else {
            answer++;
            if (currentProcess.index === location) {
                return answer;
            }
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque([(index, priority) for index, priority in enumerate(priorities)])
    
    while queue:
        index, current_priority = queue.popleft()
        
        has_higher_priority = False
        
        for process in queue:
            if process[1] > current_priority:
                has_higher_priority = True
                break 
        
        if has_higher_priority:
            queue.append((index, current_priority))
        else:
            answer += 1
            if index == location:
                return answer